plugins/experimental/cookie_remap/strip.cc

```diff
@@ -228,7 +228,7 @@ get_stripped(const char *in, ssize_t in_len, char *out, int *out_len, unsigned i
   retval = stripped_core(r, in_end, &w, out_end, flags);
 
   /* null terminate */
-  out_end += out_end ? 1 : 0;            /* undo decrement at start */
+  out_end++;                             /* undo decrement at start */
```

Comment:
A simple enough change. But for my benefit, can you explain why out_end is always non-zero? It wasn't obvious to me reading the function that it would be. At 211 it's set to:

```cpp
out_end = out + *out_len;
```

I think char is signed with some compilers, so one or both of those could be negative, which complicates things.

Reply:
maybe always was misleading but `get_stripped` only gets called here [`CookieJar::verify_value`] (https://github.pie.apple.com/ats/trafficserver/blob/master/plugins/experimental/cookie_remap/cookiejar.cc#L178) with the above code setting `out_len` > 0

Reply:
OK, so this isn't an issue with your patch, but I'm concerned this code has a one byte buffer overrun potential:

1. The caller passes in a pointer to a buffer that will be filled called `out` and a length for the allocated space called `out_len`.
2. Then we set `out_end` accordingly: `out_end = out + *out_len;`
3. OK, so out_end points to the end of the allocated buffer.
4. Then we increment one past the end of that buffer: `out_end++;`
5. And then write into the `out` buffer pointed to by `w`.

The fact is, we are probably always well under `out_len`, so this doesn't cause problems. But isn’t that a one byte buffer overrun waiting to happen? 

In other words, the caller said "Don't write more than `out_len` bytes to `out`." Then we say, "OK, I'll write at most `out_len + 1` bytes to `out`.

Let me know whether I'm missing something.

Reply:
@serrislew pointed out that earlier, in line 202, `out_len` was decremented, so this increment for `\0` should be safe per the incoming buffer.